### DISCUSSION TITLE
"Simulation failed with status: MODULE_ADDRESS_DOES_NOT_MATCH_SENDER" Getting this error when publishing on testnet

### DISCUSSION THREAD

**Discussion by ayushagarwal27** [aptos-labs/aptos-developer-discussions#364] [Questions] [answered]
### Describe your question in detail.

The contract works fine with --dev, however when that flag is removed, this kind of error is appearing

`"Simulation failed with status: MODULE_ADDRESS_DOES_NOT_MATCH_SENDER"`

### What have you tried or looked at?  Or how can we reproduce the error?

Have compiled and tried publishing code with same wallet address of testnet

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

### Describe your environment or tooling in detail

using boiler-plate-template of dApp

**Comment by gregnazario** [comment, marked as answer]
## What does `MODULE_ADDRESS_DOES_NOT_MATCH_SENDER` mean?

It means that the account publishing the module via a transaction, is not the same as the address in the module.  This is usually seen as a named address, or a plain address:

```move
module deploy_addr::my_module {
  // ...
}
```
or
```move
module 0x12345::my_module {
  // ...
}
```

In this case, it probably means that there's a dev address filled in for the `dev-addresses`

### If you have a TOML like this, the address will change based on the `--dev` flag
```toml
[package]
name = "MyPackage"
version = "1.0.0"
authors = []

[addresses]
deploy_addr = "0x55555" (or "_")

[dev-addresses]
deploy_addr = "0x12345"
```

With a header in the module like:

```move
module deploy_addr::my_module {
  // ...
}
```

### To deploy a contract you'll want to ensure that the address for `deploy_addr` is the same as your deploying account

When you deploy you can run a command like below, assuming the CLI profile `default`
```bash
aptos move publish --named-addresses deploy_addr=default
```

Or if you know the address, you can also give the address directly like so

```bash
aptos move publish --named-addresses deploy_addr=0x1234567
```

**Reply by ayushagarwal27** [reply]
Hi, thank you for your response.

I have tried the above by running from CLI

`aptos move publish --package-dir move  --skip-fetch-latest-git-deps --named-addresses poke_addr=0x12345`

move.toml have this structure

`
[addresses]
poke_addr = "0x12345"

[dev-addresses]
poke_addr = "0x12"
`

Also tried running compile first and then publish, getting same error:

`{
  "Error": "Simulation failed with status: MODULE_ADDRESS_DOES_NOT_MATCH_SENDER"
}
`

**Comment by ayushagarwal27** [comment]
Fixed it using Remix IDE
